bench-vortex/src/bin/tpcds.rs

```diff
@@ -0,0 +1,224 @@
+        .targets
+        .iter()
+        .map(|t| t.format())
+        .unique()
+        .collect_vec();
+
+    let duckdb_resolved_path = ddb::build_and_get_executable_path(&args.duckdb_path, args.fast);
+
+    for format in formats {
+        let opts = DuckdbTpcOptions::default()
+            .with_scale_factor(1)
+            .with_base_dir("tpcds".to_data_path())
+            .with_dataset(TpcDataset::TpcDs)
+            .with_duckdb_path(duckdb_resolved_path.clone())
+            .with_format(format);
+        generate_tpc(opts).expect("gen tpch-ds");
+    }
+
+    let url = Url::parse(
+        ("file:".to_owned()
+            + "tpcds"
+                .to_data_path()
+                .to_str()
+                .vortex_expect("path should be utf8")
+            + "/")
+            .as_ref(),
+    )?;
+
+    let runtime = new_tokio_runtime(None);
+
+    runtime.block_on(bench_main(
+        args.queries,
+        args.exclude_queries,
+        args.iterations,
+        1,
+        args.targets,
+        args.display_format,
+        url,
+        &duckdb_resolved_path,
+    ))?;
+
+    // Require trace guard lives until here
+    #[cfg(feature = "tracing")]
+    let _ = _trace_guard;
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn bench_main(
+    queries: Option<Vec<usize>>,
+    exclude_queries: Option<Vec<usize>>,
+    iterations: usize,
+    scale_factor: usize,
+    targets: Vec<Target>,
+    display_format: DisplayFormat,
+    url: Url,
+    duckdb_resolved_path: &Path,
+) -> anyhow::Result<()> {
+    info!(
+        "Benchmarking against these targets: {}.",
+        targets.iter().join(", ")
+    );
+
+    let mut measurements: Vec<QueryMeasurement> = Vec::new();
+    let tpch_queries: Vec<_> = tpcds_queries()
+        .filter(|(query_idx, _)| {
+            // Include query if:
+            // 1. No specific queries were requested OR this query is in the requested list
+            // 2. AND this query is not in the excluded list
+            queries
+                .as_ref()
+                .is_none_or(|included| included.contains(query_idx))
+                && exclude_queries
+                    .as_ref()
+                    .is_none_or(|excluded| !excluded.contains(query_idx))
+        })
+        .collect();
+    assert!(!tpch_queries.is_empty(), "No queries to run");
+
+    let progress = ProgressBar::new((tpch_queries.len() * targets.len()) as u64);
+
+    for target in &targets {
+        let engine = target.engine();
+        let format = target.format();
+        match engine {
+            // TODO(joe): support datafusion
+            Engine::DuckDB => {
+                let duckdb_path = duckdb_resolved_path;
+                let temp_dir = tempdir()?;
+                let duckdb_file = temp_dir
+                    .path()
+                    .join(format!("duckdb-file-{}.db", format.name()));
+
+                let url = url
+                    .join(&format!("{}/", scale_factor))
```

Comment:
#For these simple cases I think inlining the var into format is quite nice for readability: `.join(&format!("{scale_factor}/"))`